File: DB/10월 4주차/프로세스.js
function solution(priorities, location) {
  let queue = priorities.map((priority, index) => ({
    priority,
    index,
  })); // 우선순위와 원래 인덱스를 함께 저장

  let order = 0; // 몇 번째로 실행되는지 기록

  while (queue.length > 0) {
    let current = queue.shift();

    // 나머지 큐 중에서 더 높은 우선순위가 있는지 확인
    if (queue.some((p) => p.priority > current.priority)) {
      queue.push(current); // 우선순위가 더 높은 프로세스가 있으면 다시 큐의 뒤로 보냄
    } else {
      order++;
      if (current.index === location) {
        return order;
      }
    }
  }
}

File: JY/10월 5주차/프로세스.java
import java.util.*;

class Solution {
    public static class Process{
        int idx;
        int priority;

        public Process(int idx, int priority){
            this.idx = idx;
            this.priority = priority;
        }
    }
    public int solution(int[] priorities, int location) {
        // 미리 정렬 하기 
        List<Integer>arrList = new ArrayList<>();
        Queue<Process> q = new LinkedList<>();
        for(int i=0; i<priorities.length; i++){
            arrList.add(priorities[i]);
            q.add(new Process(i, priorities[i]));
        }
        /*
        [1, 1, 9, 1, 1 ,1]
        [9, 1, 1, 1, 1 ,1]
        */
                
        arrList.sort((a,b ) -> b-a);

        int answer = 0;
        
        // 큐를 돌면서 가장 큰 수부터 빼주고 아니면 뒤에 삽입
        while (!q.isEmpty()){
            Process p = q.poll();

            if(p.priority == arrList.get(answer)){
                answer++;
                if(p.idx == location){
                    break;
                }
                continue;
            }
            q.add(p);
        }
        return answer;
    }
}